File: backend/RealEstate.Domain/Entities/Property.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace RealEstate.Domain.Entities
{
    public class Property
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }

        [BsonElement("name")]
        public string Name { get; set; } = string.Empty;

        [BsonElement("address")]
        public string Address { get; set; } = string.Empty;

        [BsonElement("price")]
        public decimal Price { get; set; }

        [BsonElement("codeInternal")]
        public string CodeInternal { get; set; } = string.Empty;

        [BsonElement("year")]
        public string Year { get; set; } = string.Empty;

        [BsonElement("idOwner")]
        public string IdOwner { get; set; } = string.Empty;
    }
}

File: frontend/src/features/properties/types/Property.ts
import type { PropertyImageBase64 } from "./PropertyImage";

export default interface Property {
  id: string;
  name: string;
  address: string;
  price: number | null;
  year: number | null;
  idOwner: string;
  images: PropertyImageBase64[];
  imageEnabled: string;
};